python/lsst/meas/extensions/gaap/_gaap.py

```diff
@@ -562,16 +541,23 @@ def _gaussianizeAndMeasure(self, measRecord: lsst.afw.table.SourceRecord,
         # scaling factors and sigmas.
         if measRecord.getFootprint().getArea() == 0:
             self._setFlag(measRecord, self.name, "no_pixel")
-            raise NoPixelError
+            self._setScalingAndSigmaFlags(measRecord, self.config.scalingFactors)
+            raise NoPixelError("No good pixels in footprint", 1)
```

Comment:
If we are moving this around, let's move this below the next block that raises the `FatalAlgorithmError`. On the very off chance something went so bad in processing that all footprints have zero area and no PSF available, I'd want it to fail with a `FatalAlgorithmError`.